Realm/Realm/Sync/SessionConnectionState.cs

```diff
@@ -0,0 +1,41 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+namespace Realms.Sync
+{
+    /// <summary>
+    /// The current connection state of a sync session object.
+    /// </summary>
+    public enum SessionConnectionState : byte
```

Comment:
Should we rename this to `ConnectionState`? I don't think we need the `Session` prefix as the fact that it's an enum in the `Realms.Sync` namespace already implies it's the sync connection.

Reply:
It makes sense as the summary on the enum should clear any possible confusion.